pkg/specification/macspec/specification.go

```diff
@@ -0,0 +1,242 @@
+}
+
+// UseTxParamSetup reports whether v is allowed to use the
+// TxParamSetup{Req|Ans} MAC commands.
+func UseTxParamSetup(v ttnpb.MACVersion) bool {
+	return compareMACVersion(v, ttnpb.MACVersion_MAC_V1_0_2) >= 0
+}
+
+// UseADRParamSetup reports whether v is allowed to use the
+// ADRParamSetup{Req|Ans} MAC commands.
+func UseADRParamSetup(v ttnpb.MACVersion) bool {
+	return compareMACVersion(v, ttnpb.MACVersion_MAC_V1_1) >= 0
+}
+
+// UseRejoinParamSetup reports whether v is allowed to use the
+// RejoinParamSetup{Req|Ans} MAC commands.
+func UseRejoinParamSetup(v ttnpb.MACVersion) bool {
+	return compareMACVersion(v, ttnpb.MACVersion_MAC_V1_1) >= 0
+}
+
+// UseDeviceModeInd reports whether v is allowed to use the
+// DeviceModeInd MAC command.
+func UseDeviceModeInd(v ttnpb.MACVersion) bool {
+	return compareMACVersion(v, ttnpb.MACVersion_MAC_V1_1) >= 0
+}
+
+// UseRekeyInd reports whether v is allowed to use the
+// RekeyInd{Conf} MAC commands.
+func UseRekeyInd(v ttnpb.MACVersion) bool {
+	return compareMACVersion(v, ttnpb.MACVersion_MAC_V1_1) >= 0
+}
+
+// RekeyPeriodVersion returns the version that should be assumed
+// after a device joins in order to decrypt the RekeyInd and
+// encrypt the RekeyIndConf MAC commands.
+func RekeyPeriodVersion(v ttnpb.MACVersion) ttnpb.MACVersion {
+	return ttnpb.MACVersion_MAC_V1_1
+}
+
+// NegotiatedVersion returns the MAC version and minor version
+// that should be used by the end device and network server
+// as part of the RekeyInd{Conf} handshake.
+// v is the Network Server requested MAC version, while upperBound
+// represents the maximum minor accepted by the end device.
+func NegotiatedVersion(v ttnpb.MACVersion, upperBound ttnpb.Minor) (ttnpb.MACVersion, ttnpb.Minor) {
+	// As there is a singular minor currently available for this mechanism,
+	// we return static values.
+	return ttnpb.MACVersion_MAC_V1_1, ttnpb.Minor_MINOR_1
+}
+
+// AllowDuplicateLinkADRAns reports whether v is allowed to use
+// duplicate LinkADRAns MAC responses within the same message.
+func AllowDuplicateLinkADRAns(v ttnpb.MACVersion) bool {
+	return compareMACVersion(v, ttnpb.MACVersion_MAC_V1_0_2) >= 0 && compareMACVersion(v, ttnpb.MACVersion_MAC_V1_1) < 0
+}
+
+// AtomicLinkADRAns reports whether v will accept or reject all
+// channel masks in a singular LinkADRAns MAC response.
+func AtomicLinkADRAns(v ttnpb.MACVersion) bool {
+	return compareMACVersion(v, ttnpb.MACVersion_MAC_V1_1) >= 0
+}
+
+// LimitConfirmedTransmissions reports whether v must limit
+// the number of confirmed uplink transmissions.
+func LimitConfirmedTransmissions(v ttnpb.MACVersion) bool {
+	return compareMACVersion(v, ttnpb.MACVersion_MAC_V1_0_4) < 0
+}
+
+// RequireDevEUIForABP reports whether v requires ABP devices to have a DevEUI associated.
+func RequireDevEUIForABP(v ttnpb.MACVersion) bool {
+	return compareMACVersion(v, ttnpb.MACVersion_MAC_V1_0_4) >= 0 && compareMACVersion(v, ttnpb.MACVersion_MAC_V1_1) < 0
```

Comment:
Let's also require this for LoRaWAN 1.1. I'm aware it's not with that many words in the spec, but let's please require this.